### PR TITLE
Update the WooPayments and Woo Subscriptions deactivate plugin warning modals

### LINKED ISSUE #7240: Update the deactivate Woo Subscriptions plugin notice
### Describe the bug

When you deactivate the Woo Subscriptions plugin, the following notice is shown if:

1. The new Stripe Billing feature is enabled or,
2. The store has at least 1 Stripe Billing subscription. 

<p align="center">
  
</p>

> [!NOTE]
> This logic is not by original design. With the recent change to load Stripe Billing integration code when the store has enabled the feature, has at least 1 stripe billing or is using the WCPay Subscriptions feature means that this notice is shown when deactivating Woo Subscriptions when it shouldn't. 

> [!IMPORTANT]
> **It use to show if you deactivated Woo Subscriptions and had the WCPay Subscriptions feature enabled.** 

This notice needs to be updated to:

1. Show when the store has Stripe Billing Subscriptions. 
2. Refer to the dangers of deactivating Woo Subscriptions while there are Stripe Billing subscriptions.

There's also a weird bug if you have WooCommerce.com connected, clicking the deactivate button doesn't show the notice because the button has an ID of `deactivate-woocommerce-com-woocommerce-subscriptions`. Which means the [code which intercepts](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/subscriptions/assets/js/plugin-page.js#L53) the deactivate button click doesn't work. 

### To Reproduce

1. Install Woo Subscriptions. 
2. Enable Stripe Billing. 
3. Purchase a subscription using Stripe Billing. 
4. Deactivate Woo Subscriptions.

You should see a notice that warns about the risks of that.

### PR DESCRIPTION
**Pull request by james-allan**
Fixes #7240 

### Changes proposed in this Pull Request
Prior to this PR there were two notices shown if you disabled WooPayments or Woo Subscriptions. 

If you **deactivated Woo Subscriptions**, it would warn you that you'll start using the built-in WCPay subscriptions feature. Given this feature is being sunset, this notice needed to be updated. 

| Before | After |
|--------|--------|
| Screenshot 2023-09-29 at 1 47 12 pm | Screenshot 2023-09-29 at 1 47 12 pm | 

Unlike the previous notice, this notice is shown only if the store has active Stripe Billing subscriptions. Because "downgrading" to WCPay Subscriptions is no longer an option, the notice content has been updated to express the issues when deactivating the Woo Subscriptions plugin while there are off-site billing Stripe Billing subscriptions running. 

----

When you **deactivate WooPayments** there are two different notices that will be shown. 

1. If Woo Subscriptions is active and you have active Stripe Billing subscriptions you will see this notice:

Screenshot 2023-09-29 at 3 52 53 pm

This is essentially a duplicate of the Woo Subscriptions deactivate notice (above) but refers to WooPayments instead. 

2. If you don't have Woo Subscriptions (ie you are a legacy WCPay Subscriptions built-in subscriptions feature user), you will see this slightly different notice tailored to that wcpay subscriptions user context. 

Screenshot 2023-09-29 at 3 17 19 pm

#### Info: 

- The "off-site billing engine" links send the user to [this doc](https://woocommerce.com/document/woopayments/built-in-subscriptions/comparison/#billing-engine). 
- The "cancel these subscriptions" link sends users to [this doc](https://woocommerce.com/document/subscriptions/store-manager-guide/#cancel-or-suspend-subscription) on how to cancel and suspend subscriptions.
- Only the WCPay related notice has a learn more link. Its sends the user to [this doc](https://woocommerce.com/document/woopayments/built-in-subscriptions/deactivate/#existing-subscriptions).

@aheckler would you mind giving these notices a look and provide any feedback on the notice content and potential links, updates. :) 

### Testing instructions

**No Subscriptions.** 

1. Go to the Admin Plugins screen. 
2. Deactivate Woo Subscriptions.
   - Note no notice is shown.
3. Deactivate WooPayments.
   - Note no notice is shown.

**With active Stripe Billing subscriptions + Woo Subscriptions**

1. Go to the Admin Plugins screen. 
2. Click deactivate Woo Subscriptions. You should see the following notice: 

Screenshot 2023-09-29 at 4 24 47 pm

3. Click cancel. Test that the notice is dismissed.
4. Click "Yes, deactivate Woo Subscriptions" 
5. Confirm Woo Subscriptions is deactivated. 
6. Reactivate Subscriptions. 
7. Deactivate WooPayments. You should see the following notice: 

Screenshot 2023-09-29 at 4 27 19 pm

8. Click cancel. Test that the notice is dismissed.
9. Click "Yes, deactivate Woo Subscriptions" 
10. Confirm Woo Subscriptions is deactivated. 
11. Reactivate WooPayments. 

**With active Stripe Billing subscriptions without Woo Subscriptions**

1. Enable the WCPay Subscriptions feature. (via the option directly or via the WCPay dev tools)
1. Reactivate Subscriptions. 
2. Deactivate WooPayments. 

You should see this WCPay subscriptions centric notice: 

Screenshot 2023-09-29 at 5 51 26 pm

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7323` or branch name `issue/7240` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=issue/7240) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 1b94215231d1f1b5d7c0781ee862f75a937afcf9
- Build time: 2023-10-04 01:33:15 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by aheckler** [comment]
LGTM!

File: includes/subscriptions/class-wc-payments-subscriptions-plugin-notice-manager.php [hunk 1]
```diff
@@ -77,8 +71,16 @@ public function enqueue_scripts_and_styles() {
...
-			wc_get_template( 'html-subscriptions-plugin-notice.php', [], '', dirname( __DIR__ ) . '/subscriptions/templates/' );
+		if ( ! $this->is_admin_plugins_screen() ) {
+			return;
+		}
+
+		wc_get_template( 'html-subscriptions-plugin-notice.php', [], '', dirname( __DIR__ ) . '/subscriptions/templates/' );
+
+		// Load a slightly different notice for folks still using the legacy WCPay Subscriptions functionality.
+		if ( WC_Payments::get_gateway()->is_subscriptions_plugin_active() ) {
+			wc_get_template( 'html-woo-payments-deactivate-warning.php', [], '', dirname( __DIR__ ) . '/subscriptions/templates/' );
```
Comment (mattallan) [inline includes/subscriptions/class-wc-payments-subscriptions-plugin-notice-manager.php:82]: When I try to test deactivating WooPayments with Woo Subscriptions, the deactivate button doesn't do anything and I can see the following console log:

image

After debugging, I found the html-woo-payments-deactivate-warning.php template file doesn't exist. Do you have it locally still?

Comment (james-allan) [reply includes/subscriptions/class-wc-payments-subscriptions-plugin-notice-manager.php:82]: ah thanks. I forgot to submit one of the new templates for one of the modals. I've fixed it in https://github.com/Automattic/woocommerce-payments/pull/7323/commits/1b94215231d1f1b5d7c0781ee862f75a937afcf9

**Review by mattallan — APPROVED** [review]
LGTM @james-allan I've tested both of the flows including:
- [x] Deactivating WooPayments with active stripe billing subscriptions and no Woo Subscriptions.
- [x] Deactivating WooPayments with active stripe billing and Woo Subscriptions still active.

Code looks good.

**Pull request merged**